tests/test_middleware_interface.py

```diff
@@ -76,8 +75,22 @@ def fake_file_data(filename, dimensions, instrument):
 class MiddlewareInterfaceTest(unittest.TestCase):
     """Test the MiddlewareInterface class with faked data.
     """
+    @classmethod
+    def setUpClass(cls):
+        cls.env_patcher = unittest.mock.patch.dict(os.environ,
+                                                   {"IP_APDB": "localhost"})
```

Comment:
Why is this patch needed? Is `make_pgpass` somehow being called in the test environment? It wasn't before, and shouldn't be. 😨

Reply:
No, `middleware_interface` needs the env variable when it sets up the pipeline (currently at https://github.com/lsst-dm/prompt_prototype/blob/501d70450f88f816cff9c39ffbc4c7f548092ccc/python/activator/middleware_interface.py#L112).  If @parejkoj moves the pipeline initialization we might be able to patch only on the relevant tests (`prep_butler`) and could avoid the env_patching which is needed when the import happens at the top level of `middleware_interface.py`. 

Apologies I didn't think of moving it earlier.